File: backend/app/schemas/exchange.py
from pydantic import BaseModel
from typing import Optional, List
from datetime import datetime

class ExchangeBase(BaseModel):
    title: str
    description: Optional[str] = None
    status: str = "pending"
    relinquished_property_id: int
    replacement_property_ids: List[int] = []
    exchange_period_days: int = 45
    identification_period_days: int = 45

class ExchangeCreate(ExchangeBase):
    pass

class ExchangeUpdate(ExchangeBase):
    pass

class Exchange(ExchangeBase):
    id: int
    owner_id: int
    created_at: datetime
    updated_at: Optional[datetime] = None

    class Config:
        from_attributes = True 

File: frontend/src/types/exchange.ts
import { Property } from './property';

export type ExchangeStatus = 'pending' | 'active' | 'completed' | 'cancelled';

export interface ExchangeBase {
  relinquished_property_id: number;
  replacement_property_id: number;
  status: ExchangeStatus;
  notes?: string;
}

export interface ExchangeCreate extends ExchangeBase {}

export interface ExchangeUpdate extends Partial<ExchangeBase> {}

export interface Exchange extends ExchangeBase {
  id: number;
  owner_id: number;
  created_at: string;
  updated_at: string;
  relinquished_property?: Property;
  replacement_property?: Property;
} 